src/vtok_agent/src/agent/mod.rs

```diff
@@ -194,6 +196,34 @@ impl PostSyncAction {
             });
     }
 
+    fn restart_nginx(wait_ms: u64) {
+        info!("Restarting NGINX.");
+
+        if let Ok(pid) = is_service_running("nginx.service") {
+            debug!("Sending SIGWINCH to PID={}", pid);
+            signal::kill(unistd::Pid::from_raw(pid), signal::Signal::SIGWINCH).unwrap_or_else(
+                |err| {
+                    error!("Error sending SIGWINCH: {:?}", err);
+                },
+            );
+
+            std::thread::sleep(Duration::from_millis(wait_ms));
+
+            debug!("Sending SIGQUIT to PID={}", pid);
+            signal::kill(unistd::Pid::from_raw(pid), signal::Signal::SIGTERM).unwrap_or_else(
+                |err| {
+                    error!("Error sending SIGTERM: {:?}", err);
+                },
+            );
+        } else {
+            info!("NGINX service is not running");
+        }
```

Comment:
Do we actually need to send those signals if we're issuing a service restart anyway?

Reply:
The first signal is still essential for gracefully shutting down worker processes. This approach, in my opinion, offers a better user experience.

Reply:
I also have doubts about the necessity of the second signal. I have removed it and pushed a new revision.